File: backend-apps/backend/src/main/java/com/shaderock/lunch/backend/feature/details/type/Role.java
package com.shaderock.lunch.backend.feature.details.type;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum Role {
  USER("USER"),
  EMPLOYEE("EMPLOYEE"),
  COMPANY_ADMIN("COMPANY_ADMIN"),
  SUPPLIER("SUPPLIER"),
  SYSTEM_ADMIN("SYSTEM_ADMIN");

  @Getter
  private final String name;

  public String getNameWithPrefix() {
    return "ROLE_" + name;
  }
}

File: android/BLPMini/domain/src/main/java/com/shaderock/lunch/blpmini/domain/entity/Role.kt
package com.shaderock.lunch.blpmini.domain.entity

enum class Role(val roleName: String) {
    USER("USER"),
    EMPLOYEE("EMPLOYEE"),
    COMPANY_ADMIN("COMPANY_ADMIN"),
    SUPPLIER("SUPPLIER"),
    SYSTEM_ADMIN("SYSTEM_ADMIN");
}